astroid/inference.py

```diff
@@ -814,6 +823,22 @@ def _get_binop_flow(
             _bin_op(left, binary_opnode, op, right, context),
             _bin_op(right, binary_opnode, op, left, reverse_context, reverse=True),
         ]
+
+    if (
+        PY310_PLUS
```

Comment:
Isn't it py310 or future annotations used ?

Reply:
I don't think so. AFAIK inference should model the runtime effects and for Python 3.9 `int | str` would throw a `TypeError`.

We do exclude handle `int | str` in typing contexts with `from __future__ import annotations` separately, in pylint.